src/Compilers/CSharp/Test/Syntax/Parsing/AwaitParsingTests.cs

```diff
@@ -326,6 +326,124 @@ void f()
             }
         }
 
+        [Fact]
+        public void NestedAnonymousMethodAwait()
```

Comment:
noticed this missing test.  added just to ensure the expected flowing of 'async' into a nested non-async delegate.